tmt/result.py

```diff
@@ -302,31 +322,54 @@ def interpret_result(self, interpret: ResultInterpret) -> 'Result':
         :returns: :py:class:`Result` instance containing the updated result.
         """
 
-        if interpret in (ResultInterpret.RESPECT, ResultInterpret.CUSTOM):
+        if interpret == ResultInterpret.CUSTOM:
+            return self
+
+        # Check for failed checks
+        checks_failed = any(check.result == ResultOutcome.FAIL for check in self.check)
+
+        if interpret == ResultInterpret.RESPECT:
+            if self.respect_checks and checks_failed:
```

Comment:
`respect_check` feels more like an input, IIUIC it's basically `interpret`, but in respect to checks. So, they should enter the method through the same door, so to speak.

Reply:
I see, so you want me to create
```
class CheckResults(enum.Enum):
    RESPECT = 'respect'
    IGNORE = 'ignore'
...
```
?

I'm so lost in all the schemas, specs, et al.

Reply:
I suppose we could re-use `ResultInterpret`, it has the same purpose, and `skip` can serve as `ignore`.

Reply:
Would this help?

```python
return _result.interpret_result(
    ResultInterpret(invocation.test.result) if invocation.test.result else ResultInterpret.RESPECT,
    ResultInterpret(invocation.test.check_result) if invocation.test.check_result else ResultInterpret.RESPECT
)
```

Reply:
> I'm so lost in all the schemas, specs, et al.

It shouldn't be that messy. If I can help somehow, let me know, maybe the whole area needs more documentation.

Reply:
> ```python
> return _result.interpret_result(
>     ResultInterpret(invocation.test.result) if invocation.test.result else ResultInterpret.RESPECT,
>     ResultInterpret(invocation.test.check_result) if invocation.test.check_result else ResultInterpret.RESPECT
> )
> ```

Trying to wrap my head around it. Something like this: 
```
diff --git a/tmt/result.py b/tmt/result.py
index 7e172a47..c8ab2238 100644
--- a/tmt/result.py
+++ b/tmt/result.py
@@ -237,10 +237,10 @@ class Result(BaseResult):
         serialize=lambda path: None if path is None else str(path),
         unserialize=lambda value: None if value is None else Path(value)
         )
-    check_result: str = field(
-        default='respect',
-        serialize=lambda value: value,
-        unserialize=lambda value: value
+    check_result: ResultInterpret = field(
+        default=ResultInterpret.RESPECT,
+        serialize=lambda value: value.value,
+        unserialize=lambda value: ResultInterpret(value)
         )
 
     @classmethod
@@ -286,7 +286,8 @@ class Result(BaseResult):
         default_ids.update(ids)
         ids = default_ids
 
-        check_result = cast(str, invocation.test.node.get('check-result', 'respect'))
+        check_result = ResultInterpret(invocation.test.node.get(
+            'check-result', ResultInterpret.RESPECT.value))
 
         _result = Result(
             name=invocation.test.name,
@@ -319,6 +320,7 @@ class Result(BaseResult):
         attributes, following the ``interpret`` value.
 
         :param interpret: how to interpret current result.
+        :param check_interpret: how to interpret check results.
         :returns: :py:class:`Result` instance containing the updated result.
         """
 
@@ -329,7 +331,7 @@ class Result(BaseResult):
         checks_failed = any(check.result == ResultOutcome.FAIL for check in self.check)
 
         if interpret == ResultInterpret.RESPECT:
-            if self.check_result == 'respect' and checks_failed:
+            if self.check_result == ResultInterpret.RESPECT and checks_failed:
                 self.result = ResultOutcome.FAIL
                 if self.note:
                     self.note += ', check failed'
```
And rename "ignore" to "skip"?

Reply:
Yes, something like this, with one exception: I don't understand why `Result.check_result` exists. How checks should be interpreted is a test-level metadata key, i.e. it should come from the test, not the result. Isn't it the same story as `Test.result`?

Reply:
Thanks. Honestly that's a leftover from the previous attempts to make things work that I missed and kept.

Reply:
In the light of https://github.com/teemtee/tmt/issues/3185#issuecomment-2383398847, no `check-result` on test level, I suppose...

This means new `result` check-level field would reside in `tmt.checks.Check` class (and `_RawCheck`), and interpretation of check results need to match a result with corresponding check in `test.check` list...

Reply:
Yep